client/js/controllers/post_list_controller.js

```diff
@@ -44,13 +45,24 @@ class PostListController {
             enableSafety: api.safetyEnabled(),
             canBulkEditTags: api.hasPrivilege("posts:bulk-edit:tags"),
             canBulkEditSafety: api.hasPrivilege("posts:bulk-edit:safety"),
+            canBulkDelete: api.hasPrivilege("posts:bulk-edit:delete"),
             bulkEdit: {
                 tags: this._bulkEditTags,
             },
         });
         this._headerView.addEventListener("navigate", (e) =>
             this._evtNavigate(e)
         );
+        // This doesn't feel like the best solution
+        this._headerView._bulkDeleteEditor.addEventListener(
+            "deleteSelectedPosts",
+            (e) => {
+                this._evtDeleteSelectedPosts(e);
+            }
+        );
+
+        // Contains the id of each post that we want to delete (while using the bulk delete feature).
+        this._bulkEditDelete = [];
```

Comment:
We could also just use `document.querySelectorAll("... > .delete")` to manage our state, but that feels dirty.